From: Don Skidmore <donald.c.skidmore@intel.com>
Date: Tue, 12 Jul 2016 18:47:38 -0400
Subject: ixgbevf: Protect ixgbevf_reset_subtask from remove event
Patch-mainline: v4.8-rc1
Git-commit: 6e469ed03ed9b21b0c2dd46f77113a85f3e2bea6
References: bsc#1024349 FATE#321240 bsc#1024355 FATE#321241

In ixgbevf_reset_subtask We weren't verifying that the port haven't
been removed, we are with this patch.

Signed-off-by: Don Skidmore <donald.c.skidmore@intel.com>
Tested-by: Andrew Bowers <andrewx.bowers@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
+++ b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
@@ -2770,6 +2770,7 @@ static void ixgbevf_reset_subtask(struct
 
 	/* If we're already down or resetting, just bail */
 	if (test_bit(__IXGBEVF_DOWN, &adapter->state) ||
+	    test_bit(__IXGBEVF_REMOVING, &adapter->state) ||
 	    test_bit(__IXGBEVF_RESETTING, &adapter->state))
 		return;
 
